lua/java/api/runner.lua

```diff
@@ -0,0 +1,228 @@
+local log = require('java.utils.log')
+local async = require('java-core.utils.async').sync
+local get_error_handler = require('java.handlers.error')
+local jdtls = require('java.utils.jdtls')
+local notify = require('java-core.utils.notify')
+
+local DapSetup = require('java-dap.api.setup')
+
+--- @class BuiltInMainRunner
+--- @field win  number
+--- @field bufnr number
+--- @field job_id number
+--- @field is_open boolean
+local BuiltInMainRunner = {}
+
+function BuiltInMainRunner:_set_up_buffer_autocmd()
+	local group = vim.api.nvim_create_augroup('logger', { clear = true })
+	vim.api.nvim_create_autocmd({ 'BufHidden' }, {
+		group = group,
+		buffer = self.bufnr,
+		callback = function(_)
+			self.is_open = true
+		end,
+	})
+end
+
+function BuiltInMainRunner:new()
+	local o = {
+		is_open = false,
+	}
+
+	setmetatable(o, self)
+	self.__index = self
+	return o
+end
+
+--- @param clear_buffer boolean
+function BuiltInMainRunner:stop(clear_buffer)
+	if self.job_id ~= nil then
+		vim.fn.jobstop(self.job_id)
+		vim.api.nvim_buf_set_lines(
+			self.bufnr,
+			-2,
+			-1,
+			false,
+			{ 'Process successfully killed.', ' ', ' ' }
+		)
+		self.job_id = nil
+		if clear_buffer then
+			vim.api.nvim_buf_set_lines(self.bufnr, 0, -1, false, {})
+		end
+	end
+end
+
+function BuiltInMainRunner:_set_up_buffer()
+	vim.cmd('sp | winc J | res 15 | buffer ' .. self.bufnr)
+	self.win = vim.api.nvim_get_current_win()
+	vim.api.nvim_win_set_option(self.win, 'number', false)
+	vim.api.nvim_win_set_option(self.win, 'relativenumber', false)
+	vim.api.nvim_win_set_option(self.win, 'signcolumn', 'no')
+	self:_set_up_buffer_autocmd()
+	self.is_open = false
+end
+
+function BuiltInMainRunner:_on_std(_, data, _)
+	vim.api.nvim_buf_set_lines(self.bufnr, -2, -1, false, data)
+	local current_buf = vim.api.nvim_get_current_buf()
+	vim.api.nvim_buf_call(self.bufnr, function()
+		local mode = vim.api.nvim_get_mode().mode
+		if current_buf ~= self.bufnr or mode ~= 'i' then
+			vim.cmd('normal! G')
+		end
+	end)
+end
+
+--- @param cmd string[]
+function BuiltInMainRunner:run_app(cmd)
+	self:stop(true)
+	if not self.bufnr then
+		self.bufnr = vim.api.nvim_create_buf(false, true)
```

Comment:
Yes, you are right. I can solve this problem in issue  #115.

Reply:
@atm1020 Wait, I thought you said logs when debugging. I thought the output buffer opened when DAP is started. You can make the changes to current PR then. My bad

Reply:
@s1n7ax  The output buffer will open when the application starts running, but currently it only writes logs to this buffer and its unable to handle user inputs. 
if I open a terminal instance in a buffer, it can handle stdin. 
Should I make this change in this pr? Sorry for the misunderstanding.

Reply:
@atm1020 Yes use this PR